### PR TITLE
ANDROID-11495 add custom icon feedback screen

### PR DESCRIPTION
**Pull request by jdelga**
### :goal_net: What's the goal?
Add Custom Icon to FeedbackScreenView

### :construction: How do we do it?
Add a new type `CUSTOM` which allows to set a `customIcon` and `customAnimation`. If these not exist, the feedback defaults to the INFO icons.

### ☑️ Checks
- [ ] I updated the documentation, including readmes and wikis. If this is a breaking change, update [UPGRADING.md](../UPGRADING.md) to inform users how to proceed. If no updates are necessary, indicate so.
- [X] Tested with dark mode.
- [X] Tested with API 23.

### :test_tube: How can I test this?
Tested using the Catalog. The custom icon is not included in the library:

Captura de pantalla 2023-05-09 a las 12 39 31

### CODE REVIEW

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/classic/components/FeedbackScreenCatalogFragment.kt [hunk 1]
```diff
@@ -112,6 +113,11 @@ class FeedbackScreenCatalogFragment(
     private enum class Type(@FeedbackType val type: Int) {
         SUCCESS(FeedbackScreenView.TYPE_SUCCESS),
         ERROR(FeedbackScreenView.TYPE_ERROR),
-        INFO(FeedbackScreenView.TYPE_INFO)
+        INFO(FeedbackScreenView.TYPE_INFO),
+        CUSTOM(FeedbackScreenView.TYPE_CUSTOM)
     }
+
+    private fun Type.getCustomIcon(): Int? =
+        if (this == Type.CUSTOM) R.drawable.feedback_error_light else null
```
Comment (dagonco) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/classic/components/FeedbackScreenCatalogFragment.kt:121]: That's a minor but probably this condition could be avoided.
If I'm not wrong, unless type == CUSTOM, this customIcon will be never consumed by the inner view.

**Review by dagonco — APPROVED** [review]
Nice work! 🚀

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/classic/components/FeedbackScreenCatalogFragment.kt (hunk 1 above)
Comment (jdelga) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/classic/components/FeedbackScreenCatalogFragment.kt:121]: I see, I can pass the custom item although it is not going to be consumed

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/classic/components/FeedbackScreenCatalogFragment.kt [hunk 2]
```diff
@@ -67,7 +69,8 @@ class FeedbackScreenCatalogFragment(
                 firstButtonLoadingText = inputFirstButtonLoadingText.toNullableString(),
                 secondButtonText = inputSecondButtonText.toNullableString(),
                 showSecondButtonAsLink = checkBoxSecondButtonAsLink.isChecked(),
-                showLoadingInButton = checkBoxShowLoadingInButton.isChecked()
+                showLoadingInButton = checkBoxShowLoadingInButton.isChecked(),
+                customIcon = selectedType.getCustomIcon(),
```
[the remark this answers is not in the document]
Comment (jdelga) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/classic/components/FeedbackScreenCatalogFragment.kt]: In that case we would need some options for icons / animations. That is why I have used just a default one as custom

**Review by gmerinojimenez — APPROVED** [review]

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/classic/components/FeedbackScreenCatalogFragment.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (jdelga) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/classic/components/FeedbackScreenCatalogFragment.kt]: I'm going to proceed to the merge as this is for current version, we can talk about a new ticket to add different examples, but it is not necessary IMHO

[the remark this answers is not in the document]
Comment (dagonco) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/classic/components/FeedbackScreenCatalogFragment.kt]: In this case I thought this should be done in this ticket, due is totally related with the new feature you are developing. If anyone wanted to verify with the catalog that this custom behaviour could be customizable, it cannot be able.
But I see I'm late for that.

**Pull request merged**